sw/device/tests/pwm_smoketest.c

```diff
@@ -0,0 +1,104 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_gpio.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_pwm.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/testing/test_framework/ottf_utils.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static volatile const uint8_t kClocksHz[] = {20, 50, 100, 200};
+static volatile const uint8_t kDutyCycles[] = {10, 30, 50, 70, 90};
+
+static dif_pwm_config_t compute_clk_config(uint32_t pwm_clk) {
+  enum {
+    // Theres a trade off on the precision of the clock and the precision of the
+    // dutycycle, the higher the kDutyCycleResulution the more precise is the
+    // the dutycycle and less precise is the pwm frequency.
```

Comment:
If you choose to stick with decimal percentages, then I think it's worth clarifying that the rounding issues result from using percentages, please.
Also, I think the sentence "By experimentation..." was meant to specify the relevant value of kDutyCycleResolution, presumably the value of 4 as per the code?

Spelling: _resolution_ in a few places, and duty cycle (two words), please.